core/lib/spree/core/search/product_filter.rb

```diff
@@ -0,0 +1,27 @@
+module Spree
+  module Core
+    module Search
+      class ProductFilter
+        attr_reader :filters
+
+        def initialize(taxon = nil)
+          @filters = []
+
+          if taxon.nil?
+            @filters << Spree::Core::ProductFilters.all_taxons
+          else
+            # @filters << Spree::Core::ProductFilters.taxons_below(taxon)
+            ## unless it's a root taxon? left open for demo purposes
+
+            @filters << Spree::Core::ProductFilters.price_filter
+            @filters << Spree::Core::ProductFilters.brand_filter
+          end
+        end
+
+        def self.partial_path
```

Comment:
Rather than this being a static method on the class I'd really like it to be an instance, as it would allow the filter to potentially render different sidebars depending on its filter logic.